### PR TITLE
Added my Preonic keymap

### PR DESCRIPTION
**Pull request by mikethetiger**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/preonic/keymaps/mikethetiger/keymap.c [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+        case QWERTY:
+          if (record->event.pressed) {
+            set_single_persistent_default_layer(_QWERTY);
+          }
+          return false;
+          break;
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/mikethetiger/keymap.c]: `break` after `return` is dead code.

File: keyboards/preonic/keymaps/mikethetiger/keymap.c [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+        case BACKLIT:
+          if (record->event.pressed) {
+            register_code(KC_RSFT);
+            #ifdef BACKLIGHT_ENABLE
+              backlight_step();
+            #endif
+            #ifdef __AVR__
+            PORTE &= ~(1<<6);
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/mikethetiger/keymap.c]:
```suggestion
            writePinLow(E6);
```

File: keyboards/preonic/keymaps/mikethetiger/keymap.c [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+            #endif
+          } else {
+            unregister_code(KC_RSFT);
+            #ifdef __AVR__
+            PORTE |= (1<<6);
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/mikethetiger/keymap.c]:
```suggestion
            writePinHigh(E6);
```

File: keyboards/preonic/keymaps/mikethetiger/keymap.c [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+ * | Tab  |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Del  |
+ * |------+------+------+------+------+-------------+------+------+------+------+------|
+ * | Caps |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "   |
+ * |------+------+------+------+------+------|------+------+------+------+------+------|
+ * | Shift|   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  |Enter |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Brite| Ctrl | Alt  | GUI  |Lower |    Space    |Raise | Left | Down |  Up  |Right |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_preonic_grid( \
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/mikethetiger/keymap.c]: The backslash escaping is not necessary here, only for the layout `#define` in the keyboard code.

File: keyboards/preonic/keymaps/mikethetiger/keymap.c [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+          if (record->event.pressed) {
+            layer_on(_RAISE);
+            update_tri_layer(_LOWER, _RAISE, _ADJUST);
+          } else {
+            layer_off(_RAISE);
+            update_tri_layer(_LOWER, _RAISE, _ADJUST);
+          }
+          return false;
+          break;
+        case BACKLIT:
```
Comment (fauxpark) [inline keyboards/preonic/keymaps/mikethetiger/keymap.c]: Actually, I think this entire keycode can be replaced with `BL_STEP`.

File: keyboards/preonic/keymaps/mikethetiger/keymap.c (hunk 1 above)
Comment (mikethetiger) [reply keyboards/preonic/keymaps/mikethetiger/keymap.c]: I only changed the keymap around and added support for my rotary encoder. Everything else was copy and pasted from the default keymap folder for the Preonic.

**Comment by noroadsleft** [comment]
Did you mean to close this?

**Comment by mikethetiger** [comment]
> Did you mean to close this?

I remade my keymap.c file and made a new pr

**Comment by noroadsleft** [comment]
Yeah I saw that after.

For future reference, you didn't need to make a new PR. You can edit your PR title as `[WIP]` for "Work in Progress" and we won't merge it. Then you just make the additional changes and commit them, then take the [WIP] tag off.

**Comment by mikethetiger** [comment]
> Yeah I saw that after.
> 
> For future reference, you didn't need to make a new PR. You can edit your PR title as `[WIP]` for "Work in Progress" and we won't merge it. Then you just make the additional changes and commit them, then take the [WIP] tag off.

Cool,,,god to know. ;)

**Pull request closed without merging**